### DISCUSSION TITLE
Longhorn Helm Chart Upgrade

### DISCUSSION THREAD

**Discussion by antwacky** [longhorn/longhorn#6119] [General]
Hello, appreciate longhorn, fantastic software.

I have a query around helm chart upgrades, my specific case being 1.3.2 to 1.4.2 (latest release).

I see there are upgrade notes for longhorn itself, but no upgrade notes for upgrades via helm charts. I assume that the helm chart will handle any niche requirements between versions, however I'm not at all confident and very reluctant to go blind! It will take me a long time to recover if I have to roll back by e.g. restoring nodes.

I have read the longhorn 1.4.2 release notes and am not sure how to apply that to the helm chart upgrade (obviously the k8s 1.25+ requirement is obvious).

If anyone can shed some light on whether I can just go ahead and upgrade with helm, or if I need to do any manual steps, it would be very much appreciated.

Thanks

**Comment by aureq** [comment]
Hi @antwacky 

I few advice here...

- Check all the release notes from 1.3.2 to 1.4.2, not just the latest one.
- Check for any customisation you may have applied in your `values.yaml` and make sure they are applicable to the latest Helm Chart.

Personally, I didn't get much issues migrating from 1.3.x to 1.4.0. Currently running on K3S v1.25.6+k3s1 and v1.4.0 just fine.

Here is also a command I used to check all my volumes and their `Engine Image`.
```bash
kubectl get volumes -n longhorn-system  -o=custom-columns='NAME:.metadata.name,ENGINE_IMAGE:.spec.engineImage,CURRENT_IMAGE:.status.currentImage,STATE:.status.state,ROBUSTNESS:.status.robustness,PVC_NAMESPACE:.status.kubernetesStatus.namespace,PVC_NAME:.status.kubernetesStatus.pvcName,LAST_KNOWN_WORKLOAD_NAME:.status.kubernetesStatus.workloadsStatus[0].workloadName'
```

**Reply by antwacky** [reply]
Thanks, I'm wondering specifically how the [upgrade](https://longhorn.io/docs/1.4.2/deploy/upgrade/) process ties in with the helm chart.

Is it possible to just do a helm upgrade and expect it to manage itself?
